lib/graphql/static_validation/literal_validator.rb

```diff
@@ -6,10 +6,10 @@ def validate(ast_value, type)
     elsif type.kind.list? && ast_value.is_a?(Array)
       item_type = type.of_type
       ast_value.all? { |val| validate(val, item_type) }
-    elsif type.kind.scalar?
-      !type.coerce_input(ast_value).nil?
+    elsif type.kind.scalar? && !ast_value.is_a?(GraphQL::Language::Nodes::AbstractNode) && !ast_value.is_a?(Array)
```

Comment:
Yes, scalar values should be a primitive value, so it should be safe to do this validation doing static validation rather than let scalars accidentally accept non-primitives.  E.g. an object or array could be passed into a string typed input field and the coercion right now will accept it and just call `.to_s` on it during coercion.